### DISCUSSION TITLE
Column styles rules does not work when the column name contain parenthesis

### DISCUSSION THREAD

**Discussion by tsn77130** [grafana/grafana#37000] [Dashboards, Panels & Visualizations]
### What Grafana version are you using?
5.2.1

### What datasource are you using?
postgres

### What OS are you running grafana on?
debian 8

### What did you do?
Tried to define different column styles

### What was the expected result?
Different styles on column, some ones in percentage via 2 decimals, some others in local format with no decimals

### What happened instead?
If the name of the column contains a parenthesis, the rule is not applied on this column, global rule /.*/ is applied instead

**Comment by davkal** [comment]
This is intended behavior. The name field takes a regular expression, so if you like to match `node(A)` you currently need to escape the parentheses: `node\(A\)`. And I'm guessing that other users are already doing this (and are relying on the fact that this field takes a regex).

Nevertheless the UX could be improved:

- the label should already make clear that it takes a regular expression
- we could escape parenthesis for the user if they have not done so, based on the assumption that they would not use grouping (the parens' use in a regex) in the match. If we escape them, we should say so in the tooltip.

**Comment by torkelo** [comment]
That is strange in other places in grafana when we have a name match or regex field we require the regex be enclosed in / (left slash)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
